CMakeLists.txt

```diff
@@ -36,9 +49,11 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 # search for Z3
 find_package(Z3 REQUIRED)
 if (Z3_FOUND AND NOT TARGET z3::z3lib)
-    add_library(z3::z3lib IMPORTED INTERFACE)
+    add_library(z3::z3lib IMPORTED INTERFACE include/Result.h)
     set_property(TARGET z3::z3lib PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${Z3_CXX_INCLUDE_DIRS})
     set_property(TARGET z3::z3lib PROPERTY INTERFACE_LINK_LIBRARIES ${Z3_LIBRARIES})
+else ()
+    message(WARNING "Did not find Z3. Exact library and other depending target will not be available")
```

Comment:
This is not necessary since the `find_package` call specifies `REQUIRED` which automatically raises an error.